util/src/main/java/tc/oc/pgm/util/event/GeneralizedEvent.java

```diff
@@ -48,14 +50,75 @@ public void setCancelled(final boolean cancel) {
     }
   }
 
-  // TODO: Un-implement Physical and EntityAction, since they are SportPaper only
-  @Override
-  public World getWorld() {
-    return getCause() instanceof Physical ? ((Physical) getCause()).getWorld() : null;
+  public @Nullable World getWorld() throws EventException {
+    if (getCause() == null) return null;
+
+    World world = getWorldIfPresent(getCause());
+    if (world != null) return world;
+
+    throw new EventException(getCause().getEventName() + " has no associated world");
   }
 
-  @Override
-  public Entity getActor() {
-    return getCause() instanceof EntityAction ? ((EntityAction) getCause()).getActor() : null;
+  public @Nullable Entity getActor() {
+    return getActorIfPresent(getCause());
+  }
+
+  public static @Nullable Entity getActorIfPresent(Event event) {
```

Comment:
These reflection methods are going to be too expensive to run on every event. We're going to need to remember/cache the `MethodHandle` for each event class. We'll definitely need unit tests for this as well.

Reply:
Actually, instead of reflection mess, why don't we just have `instanceof` checks for the specific event sub-classes? Like `PlayerEvent`, `BlockEvent`, etc..

Reply:
My idea was that reflection would cover for non-bukkit events as well.. maybe not worth the job?

Reply:
As in events that just extended `Event` but still had a `World`/`Entity`

Reply:
Probably not worth it, just cover the Bukkit basics.

Reply:
Ok pushed changes, issue with this solution Is that it does __not__ get the `Entity` from a `MatchPlayerEvent`, since it is non-bukkit.